src/bun.js/test/jest.zig

```diff
@@ -1610,7 +1610,7 @@ inline fn createScope(
             .parent = parent,
             .file_id = parent.file_id,
             .tag = if (parent.is_skip) parent.tag else tag,
-            .is_skip = is_skip or parent.is_skip,
+            .is_skip = is_skip and parent.is_skip,
```

Comment:
This will break skip tests because if the parent scope is marked as skip, the child scope should also be skipped. These are the test failures we are seeing in `test-test.test.ts`

Reply:
Makes sense. I thought to refactor it to 2 separate flags - `is_skip` and `is_only` that will be aggregated throughout the traversal. And then we can be more precise in the decisions. 
Because the `is_skip` flag alone is not sufficient to handle this logic. Sounds 🆗  to you?

Reply:
You're right. Refactored the code to avoid the complexity around maintaining `is_skip` in `--only` and non-`--only` modes. 
The skip tests now pass